lib/sisimai/bite/email/exim.rb

```diff
@@ -0,0 +1,522 @@
+
+          if readcursor & Indicators[:'message-rfc822'] > 0
+            # After "message/rfc822"
+            if e.empty?
+              blanklines += 1
+              break if blanklines > 1
+              next
+            end
+            rfc822list << e
+
+          else
+            # Before "message/rfc822"
+            next if readcursor & Indicators[:deliverystatus] == 0
+            next if e.empty?
+
+            # This message was created automatically by mail delivery software.
+            #
+            # A message that you sent could not be delivered to one or more of its
+            # recipients. This is a permanent error. The following address(es) failed:
+            #
+            #  kijitora@example.jp
+            #    SMTP error from remote mail server after RCPT TO:<kijitora@example.jp>:
+            #    host neko.example.jp [192.0.2.222]: 550 5.1.1 <kijitora@example.jp>... User Unknown
+            v = dscontents[-1]
+
+            if cv = e.match(/\A[ ][ ]+([^ \t]+[@][^ \t]+[.][a-zA-Z]+)(:.+)?\z/) || e.match(Re1[:alias])
+              #   kijitora@example.jp
+              #   sabineko@example.jp: forced freeze
+              #
+              # deliver.c:4549|  printed = US"an undisclosed address";
+              #   an undisclosed address
+              #     (generated from kijitora@example.jp)
+              if v['recipient']
+                # There are multiple recipient addresses in the message body.
+                dscontents << Sisimai::Bite.DELIVERYSTATUS
+                v = dscontents[-1]
+              end
+              v['recipient'] = cv[1]
+              recipients += 1
+
+            elsif cv = e.match(/\A[ ]+[(]generated[ ]from[ ](.+)[)]\z/) ||
+                       e.match(/\A[ ]+generated[ ]by[ ]([^ \t]+[@][^ \t]+)/)
+              #     (generated from kijitora@example.jp)
+              #  pipe to |/bin/echo "Some pipe output"
+              #    generated by userx@myhost.test.ex
+              v['alias'] = cv[1]
+
+            else
+              next if e.empty?
+
+              if e =~ Re1[:frozen]
+                # Message *** has been frozen by the system filter.
+                # Message *** was frozen on arrival by ACL.
+                v['alterrors'] ||= ''
+                v['alterrors']  += e + ' '
+
+              else
+                if boundary00.size > 0
+                  # --NNNNNNNNNN-eximdsn-MMMMMMMMMM
+                  # Content-type: message/delivery-status
+                  # ...
+                  if cv = e.match(/\A[Rr]eporting-MTA:[ ]*(?:DNS|dns);[ ]*(.+)\z/)
+                    # Reporting-MTA: dns; mx.example.jp
+                    v['lhost'] = cv[1]
+
+                  elsif cv = e.match(/\A[Aa]ction:[ ]*(.+)\z/)
+                    # Action: failed
+                    v['action'] = cv[1].downcase
+
+                  elsif cv = e.match(/\A[Ss]tatus:[ ]*(\d[.]\d+[.]\d+)/)
+                    # Status: 5.0.0
+                    v['status'] = cv[1]
+
```

Comment:
__[rubocop]__
* Metrics/BlockNesting: Avoid more than 4 levels of block nesting. [:ref](https://github.com/bbatsov/ruby-style-guide#three-is-the-number-thou-shalt-count)

<!-- Comment by [SideCI](https://sideci.com) -->